helper/resource/testing_import_state.go

```diff
@@ -78,7 +78,7 @@ func testStepImportState(
 			// Find the existing resource
 			var oldR *terraform.ResourceState
 			for _, r2 := range old {
-				if r2.Primary != nil && r2.Primary.ID == r.Primary.ID {
+				if r2.Primary != nil && r2.Primary.ID == r.Primary.ID && r2.Type == r.Type {
```

Comment:
Calling out this change explicitly in case it breaks anything in acceptance test runs, but I think this is strictly better than the previous version unless shown otherwise. (cc @catsby)